src/dataExplorer/components/Results.tsx

```diff
@@ -82,6 +83,55 @@ const Results: FC = () => {
   const [search, setSearch] = useState('')
   const {result, status, view, setView} = useContext(ResultsContext)
   const {launch} = useContext(SidebarContext)
+  const res = useMemo(() => {
+    if (view.state === 'graph' || !search.trim() || !result?.parsed) {
+      return result?.parsed
+    }
+
+    const dupped = {
+      fluxGroupKeyUnion: [...result.parsed.fluxGroupKeyUnion],
+      resultColumnNames: [...result.parsed.resultColumnNames],
+      table: {
+        length: 0,
+        columns: Object.entries(result.parsed.table.columns).reduce(
+          (acc, [k, v]) => {
+            acc[k] = {...v, data: []}
+            return acc
+          },
+          {}
+        ),
+      },
+    }
+
+    const len = result.parsed.table.length
+    const keys = Object.keys(result.parsed.table.columns)
+    let newLen = 0,
+      ni = 0
```

Comment:
Just to check if I understand it correctly:

`newLen` is to keep track of the row number in `newCols`?
`ni` is to keep track of the row number in `oldCols`?

Reply:
> like spliting out a reference to the result.parsed.table.columns object into the oldCols variable outside of the for loop as all of those object member accessors (the .) actually have a noticeable performance hit 

Thanks for explaining it. Good to know!